src/main/java/com/commercetools/sync/commons/utils/CompletableFutureUtils.java

```diff
@@ -0,0 +1,21 @@
+package com.commercetools.sync.commons.utils;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class CompletableFutureUtils {
+
+    private static final CompletableFuture<Optional<?>> EMPTY = CompletableFuture.completedFuture(Optional.empty());
+
+    /**
+     * Returns a {@link CompletableFuture} of an empty {@link Optional}. There is no guarantee
+     * that it is a singleton.
+     * @return {@link CompletableFuture} of an empty {@link Optional}
+     * */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static <T> CompletableFuture<T> emptyOptionalCompletedFuture() {
```

Comment:
Actually, it must return `Optional<T>`, not just `<T>`

Reply:
and better return every time new instance of `CompletableFuture`, since the class is mutable:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#obtrudeValue-T-

Reply:
and add a separate test case verifying mutability, smth like:
```java
@Test
public void emptyOptionalCompletedFuture_immutable() throws ExecutionException, InterruptedException {
    CompletableFuture<Optional<Integer>> completedFuture = emptyOptionalCompletedFuture();
    completedFuture.obtrudeValue(Optional.of(42));
    assertThat(completedFuture.get()).contains(42);

    CompletableFuture<Optional<Integer>> completedFuture2 = emptyOptionalCompletedFuture();
    assertThat(completedFuture2).isNotSameAs(completedFuture);
    assertThat(completedFuture2.get()).isEmpty();
}
```